pkg/sources/fake/fake.go

```diff
@@ -15,24 +15,19 @@ const (
 
 var startTime = time.Now()
 
-func getBattery(id uint16) msgs.TelemetryData {
+func getCAN(id uint16) msgs.CAN {
 	t := time.Since(startTime)
 	fid := float64(id)
-	voltage := uint16(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
-	temperature := float32(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
+	data := uint16(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
 
-	return msgs.NewTelemetryData(&msgs.BatteryModule{
-		ID:          id,
-		Voltage:     voltage,
-		Temperature: temperature,
-	})
+	return msgs.NewCAN(id, data)
 }
 
 // GenFake generates fake data
 func GenFake(bus *pubsub.MessageBus) {
 	for {
-		for battery := uint16(1); battery <= 36; battery++ {
-			bus.Publish("battery", getBattery(battery))
+		for msg := uint16(1); msg <= 36; msg++ {
+			bus.Publish("CAN", getCAN(msg))
```

Comment:
Does this support some sort of per-device or per-message/message type model? I think we'll most likely sort by message/message type, but it might be neat to see per-device activity or something like that. Might help for debugging. :)

I'm not entirely sure how this system works (would appreciate any links or documentation I can read!), but from what I understand, the publisher would process the CAN messages coming in and decide which topic(s?) to publish the message on? Would the publisher translate the ID/data into the actual data (e.g. ID: battery voltage from 0x4, data: modules 0-3) or would that be delegated to the subscriber?